### ISSUE TITLE
eig([NaN]) works

### ISSUE THREAD

**Issue by jiahao** [JuliaLang/julia#18267]
```jl
eig(fill(NaN, 1, 1))
```

produces the result

```jl
([NaN],
1x1 Array{Float64,2}:
 1.0)
``` 

which is inconsistent with matrices of any other size, e.g.

```jl
eig(fill(NaN, 2, 2))
```

```jl
LoadError: ArgumentError: matrix contains NaNs
while loading In[11], in expression starting on line 1

 in checkfinite at linalg/lapack.jl:58
 in geevx! at linalg/lapack.jl:1834
 in eigfact! at linalg/eigen.jl:32
 in eigfact at linalg/eigen.jl:57
 in eig at linalg/eigen.jl:66
```

**Comment by mcognetta** [comment]
Hi, new to contributing to Julia.

I have narrowed down where this problem is but I have a question about how to implement a fix.

Line 31 in linalg/eigen.jl, `issymmetric(A) && return eigfact!(Symmetric(A))`, is where the problem is and the eigenfact! that is called is line 196 of linalg/symmetric.jl `eigfact!{T<:BlasReal,S<:StridedMatrix}(A::RealHermSymComplexHerm{T,S}) = Eigen(LAPACK.syevr!('V', 'A', A.uplo, A.data, 0.0, 0.0, 0, 0, -1.0)...)`. Is it better to just put an extra conditional in the eigen.jl code, 
something like `(issymmetric(A) && !((size(A) == (1,1)) && isnan(A[1])) && return eigfact!(Symmetric(A))` or should it be put into the eigfact! code?

BTW, I am using the 0.6.0-dev.642 version (I could not get 0.5.0 to work due to some libssh2 issue) but I think that it shouldn't matter in this case.

Thanks in advance.

**Comment by mcognetta** [comment]
One more note, it could be in the `issymmetric` function. As issymmetric returns true in the 1x1 case but false in the 2x2+ case(s).

**Comment by andreasnoack** [comment]
Fixed by #18526
